src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -43,11 +50,41 @@ const noopHandler = () => {};
 
 const ChatsList: React.FunctionComponent = () => {
   const { user } = useUser();
-  const privateChats = useSelector(privateChatsSelector);
+  const privateChats = useSelector(privateChatsSelector) ?? [];
   const chatUsers = useSelector(chatUsersSelector) ?? {};
 
   const [selectedUser, setSelectedUser] = useState<WithId<User>>();
 
+  const chats = [...privateChats];
+
+  const chatUserIds = chats
+    .sort(chatSort)
+    .flatMap((chat) => [chat.from, chat.to])
+    .filter(filterUniqueKeys)
+    .slice(0, NUM_CHAT_UIDS_TO_LOAD);
+
+  const chatUsersOption: WhereOptions = [DOCUMENT_ID, "in", chatUserIds];
+
+  const chatUsersQuery = [
+    {
+      collection: "users",
+      where: chatUsersOption,
+      storeAs: "chatUsers",
+    },
+  ];
+
+  const chatQuery = useCallback(() => {
+    if (!chatUsers || !Object.entries(chatUsers).length) {
+      if (hasElements(chatUserIds)) {
+        return chatUsersQuery;
+      }
+      return undefined;
+    }
+    return undefined;
+  }, [chatUserIds, chatUsers, chatUsersQuery]);
```

Comment:
Since `chatUsersQuery` seems to only be used within `chatQuery`, probably just inline it there at the `return` statement

Reply:
Also, what is the intent of `!Object.entries(chatUsers).length`?

Reply:
Also also, does this need to be a `useCallback`? I think `useMemo` would be sufficient, and would mean we don't have to call it below as a function.